diff --git a/dist/cjs/index.js b/dist/cjs/index.js
index e4b01510f37e6dff3c4b36736904cccb161c97f9..bce72f4b63c9d4737cfd4c9dc9de4bd79ca592b0 100644
--- a/dist/cjs/index.js
+++ b/dist/cjs/index.js
@@ -4253,8 +4253,7 @@ const defaultLayoutTransition = {
     duration: 0.45,
     ease: [0.4, 0, 0.1, 1],
 };
-const userAgentContains = (string) => typeof navigator !== "undefined" &&
-    navigator.userAgent.toLowerCase().includes(string);
+const userAgentContains = (string) => typeof navigator !== "undefined" && typeof navigator.userAgent !== "undefined" && navigator.userAgent.toLowerCase().includes(string);
 /**
  * Measured bounding boxes must be rounded in Safari and
  * left untouched in Chrome, otherwise non-integer layouts within scaled-up elements
